src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -1,18 +1,250 @@
+        f.debug_struct("Session").finish()
+    }
+}
+
+impl Session {
+    /// Create a new session
+    pub async fn new(
+        client: etcd_client::Client,
+        ttl: u64,
+        lease_id: i64,
+    ) -> DatenLordResult<Arc<Self>> {
+        let closed = Arc::new(AtomicBool::new(false));
+        let inner = Arc::new(SessionInner {
+            closed,
+            lease_id,
+            ttl,
+            client,
+        });
+        let (close_tx, close_rx) = mpsc::channel(1);
+
+        let inner_clone = Arc::clone(&inner);
+        TASK_MANAGER
+            .spawn(TaskName::EtcdKeepAlive, |token| async move {
+                inner_clone.keep_alive(close_rx, token).await;
+            })
+            .await
+            .with_context(|| "failed to spawn keep alive session task")?;
+
+        let session = Session { close_tx, inner };
+        Ok(Arc::new(session))
+    }
+
+    /// Get the lease id
+    #[must_use]
+    pub fn lease_id(&self) -> i64 {
+        self.inner.lease_id
+    }
+
+    /// Get the ttl
+    #[must_use]
+    pub fn ttl(&self) -> u64 {
+        self.inner.ttl
+    }
+
+    /// Get the `is_closed` flag
+    #[must_use]
+    pub fn is_closed(&self) -> bool {
+        self.inner.is_closed()
+    }
+
+    /// Try to close session
+    pub fn close(&self) {
+        match self.close_tx.try_send(()) {
+            Ok(o) => {
+                info!("close the keep alive session, {o:?}");
+            }
+            Err(e) => {
+                error!("failed to close the keep alive session, error={e:?}");
+            }
+        }
+        self.inner.close();
+    }
+}
+
+impl Drop for Session {
+    fn drop(&mut self) {
+        // Set the close flag
+        self.close();
+    }
+}
+
+/// The keepalive session inner struct
+pub struct SessionInner {
+    /// The closed flag
+    closed: Arc<AtomicBool>,
+    /// Current lease id
+    lease_id: i64,
+    /// The ttl of the lease
+    ttl: u64,
+    /// The etcd client
+    #[cfg_attr(not(debug_assertions), skip)]
+    client: etcd_client::Client,
+}
+
+impl Debug for SessionInner {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SessionInner")
+            .field("closed", &self.closed)
+            .field("lease_id", &self.lease_id)
+            .field("ttl", &self.ttl)
+            .finish_non_exhaustive()
+    }
+}
+
+impl SessionInner {
+    /// Get the closed flag
+    fn is_closed(&self) -> bool {
+        self.closed.load(Ordering::Acquire)
+    }
+
+    /// Set close flag
+    fn close(&self) {
+        self.closed.store(true, Ordering::Release);
+    }
+
+    /// Reopen close flag
+    fn reopen(&self) {
+        self.closed.store(false, Ordering::Release);
+    }
+
+    /// Keep alive the lease
+    #[allow(clippy::pattern_type_mismatch)] // Raised by `tokio::select`
+    pub async fn keep_alive(&self, mut close_rx: mpsc::Receiver<()>, token: CancellationToken) {
```

Comment:
1. keep_alive ->  keepalive
2. no need to retry on lease_keepalive failed: a) the looping logic is too complex and hard to make it right. b) Once error occur lease is likely already invalided by server, retry might be meaning less.  c) Only do retrying if you can distinguish error type etcd returned, otherwise retrying might have no end condition

Reply:
1. reference function is: etcd-client-0.11.1/src/client.rs
2. Thanks, I have fixed it.